platform/server-config/src/main/java/org/motechproject/server/config/SettingsFacade.java

```diff
@@ -215,6 +221,53 @@ public void saveRawConfig(String filename, String jsonText) {
     }
 
     /**
+     *  Allows persisting of raw JSON properties either in the database or file.
+     *
+     * @param filename json filename
+     * @param properties properties to persist
+     * @throws org.motechproject.commons.api.MotechException when I/O error occurs
+     */
+    public void saveRawConfig(String filename, Properties properties) {
+        Gson gson = new GsonBuilder().create();
+        Map<String, String> map = new HashMap<String, String>();
+        for (Entry<Object, Object> prop : properties.entrySet()) {
+            map.put((String) prop.getKey(), (String) prop.getValue());
+        }
+        saveRawConfig(filename, gson.toJson(map));
+    }
+
+    /**
+     * Allows to retrieve raw JSON data either from the database or file.
+     *
+     * @param filename Resource filename
+     * @return Raw JSON data as InputStream
+     * @throws org.motechproject.commons.api.MotechException when I/O error occurs
+     */
+    public Properties getPropertiesFromRawConfigFile(String filename) {
+        String json = new String();
+        try {
+            json = IOUtils.toString(getRawConfig(filename));
+        } catch (NullPointerException e) {
+            LOGGER.debug(e.getMessage());
+            return new Properties();
+        } catch (IOException e) {
+            LOGGER.debug(e.getMessage());
```

Comment:
1) This should not be a debug but an error
2) Don't swallow the stacktrace
3) Add a meaningful message